src/javascript/binary/livechart.js

```diff
@@ -281,7 +281,7 @@ LiveChartTick.prototype.process_data = function(point) {
             epoch: parseInt(point[1]),
             quote: parseFloat(point[2])
         };
-        if (!this.chart && !this.chart.series) {
+        if (!this.chart || !this.chart.series) {
```

Comment:
Totally better ;)
The nots may be confusing you, but currently if this.chart is undefined it
still continues evaluating second part and throws there. Or is the correct
operator. I've tested it.
On Mar 20, 2015 8:31 AM, "Raunak Kathuria" <notifications@github.com> wrote:

> In src/javascript/binary/livechart.js
> <https://github.com/binary-com/binary-static/pull/224#discussion_r26821128>
> :
>
> > @@ -281,7 +281,7 @@ LiveChartTick.prototype.process_data = function(point) {
> >              epoch: parseInt(point[1]),
> >              quote: parseFloat(point[2])
> >          };
> > -        if (!this.chart && !this.chart.series) {
> > +        if (!this.chart || !this.chart.series) {
>
> what difference it will make, i don't its better than before.. Say
> this.chart is defined and this.chart.series is not defined then it will not
> return and this.chart.series[0] will give error then. And we use these
> conditions at lot of places we need to update them also
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/binary-com/binary-static/pull/224/files#r26821128>.
>